packages/medusa/src/services/product.ts

```diff
@@ -384,11 +390,30 @@ class ProductService extends TransactionBaseService<ProductService> {
           product.type_id = (await productTypeRepo.upsertType(type))?.id || null
         }
 
+        if (
+          this.featureFlagRouter_.isFeatureEnabled(
+            SalesChannelFeatureFlag.key
+          ) &&
+          typeof salesChannels !== "undefined"
+        ) {
```

Comment:
nit(non-blocking): split into two `if`s for better readability and quicker resolution when feature flags are to be remove